tools/c7n_azure/c7n_azure/resources/generic_arm_resource.py

```diff
@@ -12,10 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from c7n_azure.constants import RESOURCE_GROUPS_TYPE
 from c7n_azure.provider import resources
+from c7n_azure.query import DescribeSource, ResourceQuery
 from c7n_azure.resources.arm import ArmResourceManager, arm_resource_types
-from c7n_azure.utils import ResourceIdParser
+
 from c7n.filters.core import Filter, type_schema
+from c7n.query import sources
+from c7n_azure.utils import ResourceIdParser, is_resource_group_id
+
+
+class GenericArmResourceQuery(ResourceQuery):
+
+    def filter(self, resource_manager, **params):
+        client = resource_manager.get_client()
+        results = [r.serialize(True) for r in client.resources.list()]
+
+        resource_groups = [r.serialize(True) for r in client.resource_groups.list()]
+        for r in resource_groups:
+            r['type'] = RESOURCE_GROUPS_TYPE
```

Comment:
Not sure if it's possible - but can this be added to the list comprehension so you only have to loop through once?

Reply:
The only way I can think of is to do this, but it will create extra dict for each item and I think it is more expensive than looping through them again

```
rg_extra_properties = {'type': RESOURCE_GROUPS_TYPE}
resource_groups = [dict(**r.serialize(True), **rg_extra_properties)
                   for r in client.resource_groups.list()]
```

Reply:
This is probably the same as update edits in place, but I don't think it is better.
```
resource_groups = [r.update({'type': RESOURCE_GROUPS_TYPE}) or r for r in 
                  [ro.serialize(True) for ro in client.resource_groups.list()]]
```